python-backend/app/api/document_manager/resources/document_manager.py

```diff
@@ -0,0 +1,85 @@
+from app.extensions import jwt, api, documents
+from ...utils.resources_mixins import UserMixin, ErrorMixin
+
+
+class DocumentManagerResource(Resource, UserMixin, ErrorMixin):
+    parser = reqparse.RequestParser()
+    
+    @jwt.requires_roles(["mds-mine-create"])
+    def post(self):
+
+        self.parser.add_argument('file', type=FileStorage, location='files', action='append')
+        self.parser.add_argument('folder', type=str, help='mine to the new document on')
+        self.parser.add_argument('pretty_folder', type=str, help='mine to the new document on')
+
+        try:
+            data = self.parser.parse_args()
+
+        except exceptions.RequestEntityTooLarge:
+            return({
+                'errors':[{
+                    'message': f'The maximum file upload size is {current_app.config["FILE_BYTE_LIMIT"]/1024/1024}MB please ensure all files are this size.',
+                }]
+            })
+
+        folder = data['folder']
+        pretty_folder = data['pretty_folder']
+        document_guid_list = []
+        errors = []
+        
+        for upload in data['file']:
+            try:
+                file_guid = uuid.uuid4()
+                original_file_name, file_extension = os.path.splitext(upload.filename)
+
+                upload_response = documents.save(upload, folder, (str(file_guid) + file_extension))
+                real_path = documents.path(upload_response)
+
+                #create the readable path by removing the guids and replacing them with the original file name and mine no
+                pretty_path = re.sub(r'\b'+folder+r'\b', pretty_folder, real_path)
+                pretty_path = re.sub(r'\b'+str(file_guid)+r'\b', original_file_name, pretty_path)
+                
+
+                document_info = DocumentManager( 
+                    document_guid = file_guid,
+                    full_storage_path = real_path,
+                    upload_date = datetime.now(),
+                    file_display_name = (original_file_name + file_extension),
+                    path_display_name = pretty_path,
+                    **self.get_create_update_dict(),
+                )
+                document_info.save()
+                document_guid_list.append(str(file_guid))
+
+            except UploadNotAllowed as e:
+                errors.append(
+                    {
+                        'message': f'The type of the file: {original_file_name + file_extension} is not allowed and must be of the types {current_app.config["TAILINGS_FILE_SET"]}',
+                    }
+                )
+                continue
+
+            except Exception as e:
+                errors.append(
+                    {
+                        'message': 'An unexpected error has occured: ' + str(e),
+                    }
+                )
+                continue
+
+        return {
+            'document_manager_guids' : document_guid_list,
+            'errors': errors,
+        }
```

Comment:
This is a really long method. I personally feel like many of our existing python methods are unnecessarily long.

Thoughts on breaking some of these pieces out? I'd be concerned with this method growing over time, as it is already around 60 lines long